documentation/Outcomes-Display.md

```diff
@@ -0,0 +1,427 @@
+Named properties on an object can be deeply nested, e.g. `user.firstName`
+
+  If the returned value is *not* an object, we can substitute the whole value into the substitution string using `${}` (i.e. empty parameter) e.g. to display any simple column's value in italics, we'd just add the substitution string `*${}*`
+
+  Note: the same result could be achieved using a complex evaluator query in the `value_expression` field, but this is just a simpler (and hopefully more readable) way to do basic substitutions.
+
+### A note about Array data
+
+If a column value is an array, then the array will be interpreted as a Markdown list, and all formatting options (plugin type, additional formatting, etc) will be applied to *each item* in the list. See below for an example.
+
+## Real use-case example
+
+Here is an `outcome_display` entry for showing Organisations:
+
+```
+{
+    "id": 3,
+    "tableName": "organisation",
+    "code": "organisation",
+    "title": "Organisations",
+    "permissionNames": null,
+    "tableViewIncludeColumns": [
+        "logoUrl",
+        "name",
+        "address"
+        ],
+    "tableViewExcludeColumns": null,
+    "detailViewIncludeColumns": [
+        "logoUrl",
+        "...",
+        "members"
+        ],
+    "detailViewExcludeColumns": [
+        "id",
+        "name",
+        "isSystemOrg"
+        ],
+    "detailViewHeaderColumn": "name",
+    "showLinkedApplications": true,
+    "priority": 1
+}
+```
+
+This layout describe how to display the `organisation` table. It will have the title "Organisations" (when in Table view).
+
+- `permissionNames` is `null`, so all users will see this layout
+- Table view will show the three named columns
+- Details view, however, is a little more complex
+  - `...` shows all columns. The reason why we include it *after* `logoUrl` is because we want the logo to appear first in the Detail table.
+  - the column name `members` is not a field on the `organisation` table. This means we will expect it to have a custom definition in the `outcome_display_column_definition` table.
+  - we exclude `id`, `name` and `isSystemOrg`, so the complete list of fields that will be shown in Details view is (in order):
+    - logoUrl
+    - registration
+    - address
+    - registration_documentation
+    - members (custom column)
+- In Details view, the page Header will be the value from the `name` field (which is why it made sense to exclude it from the list of fields to show)
+- We also are allowed to see a list of Linked Applications in Details view
+
+We now define three columns with custom definitons in `outcome_display_column_definition`: `logoUrl`, `registrationDocumentation` and `members`. That means that the other two will just use their field name as column title, and the values in those fields will be shown as plain text.
+
+Let's look at the custom column definitions one by one:
+
+**logoUrl**
+
```

Comment:
```suggestion
`outcome_display_column_definition`
```

Just so it's easier to track which is this referring to

Reply:
I see, you mean at the start of the object block.

I just tried it and it looked weird, so have added a clarifying note in the "Let's look at..."